modules/gospel.js

```diff
@@ -0,0 +1,315 @@
+  const scriptureMasteries = require("../data/gospel/scripture-mastery-reference.json");
+  const reference = u.rand(scriptureMasteries);
+  return {
+    book: reference[0],
+    chapter: reference[1],
+    verses: reference[2]
+  };
+}
+
+/**
+ * Displays a verse that's requested, or a random verse if none is specified.
+ * @param {Interaction} interaction The interaction that caused this command.
+ */
+async function slashGospelVerse(interaction) {
+  let book = interaction.options.getString("book", false);
+  let chapter = interaction.options.getInteger("chapter", false);
+  let verses = interaction.options.getString("verses", false);
+
+  if (!book || !chapter) {
+    // Get a random one from scripture mastery.
+    ({ book, chapter, verses } = getScriptureMastery());
+  }
+
+  const bookRef = abbreviationTable.get(book.toLowerCase());
+  if (!bookRef) {
+    interaction.reply({ content: "I don't understand what book you're mentioning.", ephemeral: true });
+    return;
+  }
+
+  // Parse verses.
+  let versesNums;
+  try {
+    versesNums = parseVerseRange(verses);
+  } catch (e) {
+    if (e instanceof SyntaxError) {
+      interaction.reply({ content: "I don't understand what verses you're looking for.", ephemeral: true });
+      return;
+    } else {
+      throw e;
+    }
+  }
+  // Put together the embed
+  const embed = u.embed()
+    .setTitle(bookRef.bookName + " " + chapter.toString() + (versesNums[0] ? ":" + verses : ""))
+    .setURL(`https://www.churchofjesuschrist.org/study/scriptures/${bookRef.work}/${bookRef.urlAbbrev}/${chapter}${(versesNums[0] ? ("." + verses + "?lang=eng#p" + versesNums[0]) : "?lang=eng")}`);
+  const bookJson = require("../data/gospel/" + works[bookRef.work] + "-reference.json");
+  if (!bookJson[bookRef.bookName][chapter]) {
+    interaction.reply({ content: `That chapter doesn't exist in ${bookRef.bookName}!`, ephemeral: true });
+    return;
+  }
+  const verseContent = [];
+  for (const num of versesNums) {
+    if (bookJson[bookRef.bookName][chapter][num]) {
+      verseContent.push(num.toString() + " " + bookJson[bookRef.bookName][chapter][num]);
+    }
+  }
+  const verseJoinedContent = verseContent.join("\n\n");
+  if (verses && verseJoinedContent.length === 0) {
+    interaction.reply({ content: "The verse(s) you requested weren't found.", ephemeral: true });
+    return;
+  }
+  embed.setDescription(verseJoinedContent.length > 2048 ? verseJoinedContent.slice(0, 2048) + "…" : verseJoinedContent);
+  embed.setColor(0x012b57);
+  interaction.reply({ embeds: [embed] });
+}
+
+/**
+ * Splits the verses section in a scripture reference into individual verse numbers.
+ * @param {string} verses A string containing numbers, spaces, hyphens, and commas.
+ * @returns An Array of numbered integers as interpreted. "3-5, 7" returns [3, 4, 5, 7]
+ */
+function parseVerseRange(verses) {
+  let versesNums;
+  if (verses) {
+    verses = verses.replace(/ /g, "");
+    const versesList = verses.split(",");
```

Comment:
Let's adjust this to also allow semicolons.
```suggestion
    const versesList = verses.split(/[,;]/);
```